packages/core/src/llm/executor/stream-processor.ts

```
@@ -645,3 +710,65 @@ export class StreamProcessor {
+function tryParsePartialJson(input: string): Record<string, any> | null {
+    const trimmed = input.trim();
+    if (!trimmed) return null;
+    if (!trimmed.startsWith('{') && !trimmed.startsWith('[')) return null;
+
+    // Remove trailing commas before attempting repair.
+    let repaired = trimmed.replace(/,\s*$/, '');
+
+    let openBraces = 0;
+    let openBrackets = 0;
+    let inString = false;
+    let escape = false;
```

Biome (2.3.14)

[error] Do not shadow the global "escape" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)